synchrad/calc.py

```diff
@@ -527,16 +528,29 @@ def _init_comm(self):
             self.ocl_version = "None"
         else:
             try:
-                self.ctx = cl.create_some_context(**ctx_kw_args)
+                print(f"Creating context with args: {ctx_kw_args}")  # Logging
+
+                # Set up OpenCL context
+                platforms = cl.get_platforms()
+                gpus = platforms[0].get_devices(device_type=cl.device_type.GPU)
```

Comment:
i suspect this could affect POCL and AMD platforms which implement CPU code (inefficient but still working). We meed to see how it handles these cases.

Reply:
Yes, I only tested this on an NVIDIA multi-GPU system, and do not expect it to work out of the box on CPUs. AMD might work, though (needs testing). Or we can decide to simplify and only support NVIDIA GPUs, like fbpic does.

Reply:
I do not have access to an AMD system for testing this atm, do you?